pkg/crc/machine/start.go

```diff
@@ -112,41 +112,104 @@ func (client *client) updateVMConfig(startConfig types.StartConfig, vm *virtualM
 
 func growRootFileSystem(sshRunner *crcssh.Runner, preset crcPreset.Preset) error {
 	if preset == crcPreset.Microshift {
-		logging.Debugf("growRootFileSystem does not support LVM which is used by %s images", preset)
-		return nil
+		return growLVForMicroshift(sshRunner)
 	}
 	// With 4.7, this is quite a manual process until https://github.com/openshift/installer/pull/4746 gets fixed
 	// See https://github.com/crc-org/crc/issues/2104 for details
-	rootPart, _, err := sshRunner.Run("realpath", "/dev/disk/by-label/root")
+	rootPart, err := getrootPartition(sshRunner, "/dev/disk/by-label/root")
 	if err != nil {
 		return err
 	}
+
+	if err := runGrowpart(sshRunner, rootPart); err != nil {
+		return err
+	}
+
+	logging.Infof("Resizing %s filesystem", rootPart)
+	if _, _, err := sshRunner.RunPrivileged("Remounting /sysroot read/write", "mount -o remount,rw /sysroot"); err != nil {
+		return err
+	}
+	if _, _, err = sshRunner.RunPrivileged(fmt.Sprintf("Growing %s filesystem", rootPart), "xfs_growfs", rootPart); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getrootPartition(sshRunner *crcssh.Runner, label string) (string, error) {
+	rootPart, _, err := sshRunner.Run("realpath", label)
+	if err != nil {
+		return "", err
+	}
 	rootPart = strings.TrimSpace(rootPart)
 	if !strings.HasPrefix(rootPart, "/dev/vda") && !strings.HasPrefix(rootPart, "/dev/sda") {
-		return fmt.Errorf("Unexpected root device: %s", rootPart)
+		return "", fmt.Errorf("Unexpected root device: %s", rootPart)
 	}
+	return rootPart, nil
+}
+
+func runGrowpart(sshRunner *crcssh.Runner, rootPart string) error {
+	var exitErr *ssh.ExitError
 	// with '/dev/[sv]da4' as input, run 'growpart /dev/[sv]da 4'
-	if _, _, err := sshRunner.RunPrivileged(fmt.Sprintf("Growing %s partition", rootPart), "/usr/bin/growpart", rootPart[:len("/dev/.da")], rootPart[len(rootPart)-1:]); err != nil {
-		var exitErr *ssh.ExitError
-		if !errors.As(err, &exitErr) {
-			return err
-		}
-		if exitErr.ExitStatus() != 1 {
-			return err
-		}
+	_, _, err := sshRunner.RunPrivileged(fmt.Sprintf("Growing %s partition", rootPart), "/usr/bin/growpart", rootPart[:len("/dev/.da")], rootPart[len(rootPart)-1:])
+	if !errors.As(err, &exitErr) {
+		return err
+	}
+	if exitErr.ExitStatus() != 1 {
+		return err
+	}
 
-		logging.Debugf("No free space after %s, nothing to do", rootPart)
-		return nil
+	logging.Debugf("No free space after %s, nothing to do", rootPart)
+	return nil
+}
+
+func growLVForMicroshift(sshRunner *crcssh.Runner) error {
+	rootPart, err := getrootPartition(sshRunner, "/dev/disk/by-id/lvm-pv-*")
```

Comment:
I'd have a strong preference for, for example, getting the device to use using `sudo pvs -S vg_name=rhel -o pv_name --noheadings` or some similar command.
Alternatively, you can change https://github.com/openshift/microshift/blob/bac892b5586edff5687b399475f8ad0e8ec01994/scripts/image-builder/config/kickstart.ks.template#L28 to set a well-known label (see [the --label option to part](https://docs.fedoraproject.org/en-US/fedora/f36/install-guide/appendixes/Kickstart_Syntax_Reference/#sect-kickstart-commands-part)). Then you can use this label to `getRootPartition`.

Passing a wildcard to `getRootPartition` only works because we get the root partition through SSH, which I'd rather not leak outside of `getRootPartition`. And there's nothing in the code erroring out if we are unlucky and there are multiple matches for lvm-pv-*.